src/sql/platform/accounts/common/accountViewModel.ts

```diff
@@ -48,23 +53,64 @@ export class AccountViewModel {
 		// 2) For each provider, get the accounts
 		// 3) Build parameters to add a provider and return it
 		try {
-			const metadata = await this._accountManagementService.getAccountProviderMetadata();
-			const accounts = await Promise.all(metadata.map(async providerMetadata => {
-				try {
-					const accounts = await this._accountManagementService.getAccountsForProvider(providerMetadata.id);
-					return <AccountProviderAddedEventParams>{
-						addedProvider: providerMetadata,
-						initialAccounts: accounts
-					};
-				} catch (err) {
-					this._logService.warn(`Error getting accounts for provider ${providerMetadata.id} : ${err}`);
-					return undefined;
-				}
-			}));
+			let accounts = await this.getAccountsForProviders();
+			if (this._authenticationService) {
```

Comment:
We should always have an authService (otherwise something is horribly wrong), so I would suggest just removing this check. If tests are failing because you're passing in null then you should be creating a mock service and passing that in instead.